examples/basic/mining_nar.py

```diff
@@ -0,0 +1,112 @@
+          "(DES), described by Iztok Fister et al. in \"uARMSolver: A framework for "
+          "Association Rule Mining\". It is a nature-inspired stochastic optimization "
+          "algorithm.\n")
+    print("As a demonstration of working with some of DES' parameters, let's inspect "
+          "a dataset containing information about 159 dog breeds.\n")
+    df = pandas.read_csv(TABLE)
+
+    print("Fragment of the dog_breeds.csv table:")
+    print(df)
+    print("\nA fragment of the table is presented above. In total, each dog breed has"
+          " 13 attributes.")
+    print("Now, let's mine some NARs. We will use a minimum support of 0.1 and a minimum "
+          "confidence of 0.7. Note that the minimum support and confidence only affect "
+          "which of the discovered NARs are returned, but the mining process itself "
+          "remains unchanged. We will also use a population size of 500 and "
+          "max_fitness_evaluations of 700. Larger values for max_fitness_evaluations "
+          "tend to return larger rules encompassing more attributes. The population size "
+          "parameter affects the number of NARs being generated and mutated. Larger values "
+          "are slower but output more NARs.\n")
+    algo.execute(minconf=0.7, minsup=0.1, population_size=500,
+                 max_fitness_evaluations=700)
+    if len(algo.get_nars()) != 1:
+        raise ValueError("example requires that a single NAR was mined")
+    discovered_nar = algo.get_nars()[0]
+    print_10_nars([discovered_nar], df.columns)
+    
+    print("\nThe above NAR is the only one discovered with these settings. The NAR "
+          "states that about 92% of all dog breeds of type "
+          "'Hound' have an intelligence rating between 6 and 8 out of 10 and are between "
+          "sizes 0 and 4 out of 5 (0 being 'Toy' and 5 being 'Giant'). This suggests "
+          "that, in general, hounds are intelligent dogs and no more than "
+          "8% of all hounds are of 'Giant' size. Let's see if that is true.\n")
+
+    hound_rows = df[df['Type'] == 'Hound']
+    
+    violating_row_indices = []
+    min_intelligence = discovered_nar.cons[9].lower_bound
+    max_intelligence = discovered_nar.cons[9].upper_bound
+    for i, (_, row) in enumerate(hound_rows.iterrows()):
+        intelligence = row['Intelligence']
+        if intelligence < min_intelligence or intelligence > max_intelligence:
+            violating_row_indices.append(i)
+        
+    header, *hound_row_strings = hound_rows[['Name', 'Type', 'Intelligence', 'Size']].to_string().splitlines()
+    for i, hound_row_string in enumerate(hound_row_strings):
+        if i in violating_row_indices:
+            print(f"{Back.RED}{hound_row_string}{Back.RESET}")
+        else:
+            print(hound_row_string)
+    
+    print("\nAs observed, only 2 rows with 'Type' equal to 'Hound' fall outside "
+          "the intelligence rating range of 6 to 8. These two records account for "
```

Comment:
Wait, I'm actually not sure here. What did you want to convey with "explain [the confidence level]"?